### PR TITLE
[CI] Process GitHub API rate limit error

### PR DESCRIPTION
**Pull request by itrushkin**
## Issue
CI job `test-insecure-client` fails because of exceeding the GitHub API rate limit. We fetch `step-ca` binaries from GitHub by calling `api.github.com` which has a rate limit for unauthorized clients. When we create multiple pull requests in a short period of time, some tests may fail due to GitHub request denials.

## Solution
Change step-ca URLs from `api.github.com` to `dl.step.sm`.

### CODE REVIEW

File: tests/github/pki_insecure_client.py [hunk 1]
```diff
@@ -45,9 +45,21 @@ def install_ca():
         'fx', 'pki', 'install',
         '-p', str(CA_PATH),
         '--password', str(CA_PASSWORD)
-    ], stdin=subprocess.PIPE)
+    ], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
     time.sleep(3)
-    pki_install.communicate('y'.encode('utf-8'))
+    stdout, _ = pki_install.communicate('y'.encode('utf-8'))
+    while 'API rate limit exceeded' in stdout.decode('utf-8'):
+        minutes_to_wait = 5
```
Comment (igor-davidyuk) [inline tests/github/pki_insecure_client.py]: Would it be better to move this constant to the top of the script?
Also, is there any meaning in 5 minutes delay, can we make it smaller?

**Comment by psfoley** [comment]
Marking `WIP` while @itrushkin looks into alternate download location for `step-ca`

**Comment by itrushkin** [comment]
Removed WIP

File: openfl/component/ca/downloader.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import platform
+import urllib.request
+import shutil
+
+from click import confirm
+
+VERSION = '0.16.0'
```
Comment (mansishr) [inline openfl/component/ca/downloader.py:10]: Is there a specific reason to use this version when latest versions are available?

Comment (itrushkin) [reply openfl/component/ca/downloader.py:10]: The latest `step-ca` release has a different folder structure from what we expect to have in our PKI interface. The transition could not be done smoothly and requires changes in our code that should be addressed in different PR.

**Review by mansishr — APPROVED** [review]

**Review by igor-davidyuk — APPROVED** [review]
Looks good

**Pull request merged**